src/compiler/checker.ts

```diff
@@ -3422,6 +3422,11 @@ module ts {
                 error(errorNode, Diagnostics.Excessive_stack_depth_comparing_types_0_and_1, typeToString(source), typeToString(target));
             }
             else if (errorInfo) {
+                if (errorInfo.next === undefined) {

```

Comment:
I don't really get what this check is for. When does this happen? Why don't we get the elaboration? Please document it.